plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreContainerTests.java

```diff
@@ -19,21 +19,54 @@
 
 package org.elasticsearch.repositories.s3;
 
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
 import org.elasticsearch.common.blobstore.BlobStore;
+import org.elasticsearch.common.logging.Loggers;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.ByteSizeUnit;
 import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.mocksocket.MockServerSocket;
 import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
 
 import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
 import java.util.Locale;
 
 public class S3BlobStoreContainerTests extends ESBlobStoreContainerTestCase {
+
+    private static final Logger logger = Loggers.getLogger(S3BlobStoreContainerTests.class);
+
+    private static ServerSocket socket;
+
+    // Opens a MockSocket to simulate connections to S3 checking that SocketPermissions are set up correctly
+    @BeforeClass
+    public static void openMockSocket() throws IOException {
+        socket = new MockServerSocket(0, 50, InetAddress.getByName("127.0.0.1"));
+        new Thread(() -> {
```

Comment:
This thread can leak and fail the test, I think that you need to clean it up (join on it in tear down).